hook/hook.go

```diff
@@ -8,15 +8,64 @@
 package hook
 
 import (
+	"math"
+
 	"github.com/ava-labs/avalanchego/vms/components/gas"
+	"github.com/ava-labs/libevm/common"
 	"github.com/ava-labs/libevm/core/state"
 	"github.com/ava-labs/libevm/core/types"
 	"github.com/ava-labs/libevm/params"
+	"github.com/holiman/uint256"
 
 	"github.com/ava-labs/strevm/intmath"
 	saeparams "github.com/ava-labs/strevm/params"
 )
 
+// AccountDebit includes an amount that an account should have debited,
+// along with the nonce used to debit the account.
+type AccountDebit struct {
+	Nonce  uint64
+	Amount uint256.Int
+}
+
+// Op is an operation that can be applied to state during the execution of a
+// block.
+type Op struct {
+	// Gas consumed by this operation
+	Gas gas.Gas
+	// GasPrice this operation is willing to spend
+	GasPrice uint256.Int
+	// From specifies the set of accounts and the authorization of funds to be
+	// removed from the accounts.
+	From map[common.Address]AccountDebit
+	// To specifies the amount to increase account balances by. These funds are
+	// not necessarily tied to the funds consumed in the From field. The sum of
+	// the To amounts may even exceed the sum of the From amounts.
+	To map[common.Address]uint256.Int
+}
```

Comment:
If now isn't the time/place, we can change the `uint256.Int` into `*uint256.Int`... But it is probably faster and less error prone to pass these as values (as then they act exactly like `uint64`).